### ISSUE TITLE
Will drone support a scheduler? 

### ISSUE THREAD

**Issue by iammichiel** [drone/drone#1525]
As stated on the drone landing page : `Drone aspires to be an industry-wide replacement for Jenkins.` One of the big features of Jenkins is the scheduler functionnality. Many things can trigger a build : 
- a user input (clicking a button)
- cron expression (jobs running every night, every x)
- webhooks 
- code changes

From what I have seen found so far, not all of these are currently implemented. I am especially interested in the cron expression. Is this something that is considered?

**Comment by bradrydzewski** [comment]
Hi Ian,

I do hope that drone can provide a viable alternative to individuals using Jenkins. I do not believe we need to build everything directly into drone to achieve this, but instead expose relevant endpoint for third party integration.

This philosophy is discussed a bit more in this comment, which I encourage you to read:
https://github.com/drone/drone/issues/966#issuecomment-91316397

> From what I have seen found so far, not all of these are currently implemented. I am especially interested in the cron expression. Is this something that is considered?

The strategy we are taking with integrations can be applied to tasks like cron as well. I believe cron is best implemented as a separate utility using either traditional cron or one of the open source scheduling utilities on github (you can find some with nice UIs).

You can then use one of these tools to schedule nightly or recurring builds with Drone by invoking the API. This could possible be done with command line utility using some bash-foo:

```sh
# this will rebuild head, re-running the existing build
drone build start drone/drone $(drone build last drone/drone | sed -n 1p)

# this will rebuild head, running a new build, with a new build number
drone build fork drone/drone $(drone build last drone/drone | sed -n 1p)
```

> a user input (clicking a button)

This is supported today by clicking the re-build button. This assumes that the build already exists in the Drone database, which should be the case since drone receives a hook for every commit. If a hook is not delivered you can manually re-deliver via the drone user interface.

This doesn't support building commits prior to activating drone, but it is likely commits prior to activating drone would not have a `.drone.yml` file and therefore would not be processed by the system anyway. 

> code changes

Is there a situation where code would change and GitHub would not trigger a webhook?

**Comment by bradrydzewski** [comment]
btw this is a gist that you can use (written in Go) to schedule nightly builds from the command line:
https://gist.github.com/bradrydzewski/bc6e96cb88718de51ce3

if you prefer node I'm guessing you can re-create the same thing with:
https://github.com/drone/drone-node

hopefully this helps!

**Comment by bradrydzewski** [comment]
official docs for scheduling builds http://readme.drone.io/questions/how-to-schedule-builds/

**Comment by joekohlsdorf** [comment]
Why was this closed? The doc describes a workaround and not a true solution.

1) It is unacceptable to have the scheduling configuration in a different place/layer.
2) Most team members don't have the access privileges to configure cronjobs.
3) Drone is supposd to be cloud-native, cronjobs in k8s aren't really the way to go. Not even in Docker containers.

Lets talk what it would take for real scheduling support, like in CircleCI or Jenkins or Travis or [insert any CI solution but Drone here].

**Comment by shiningmosquito** [comment]
Circle CI schedule doc. https://circleci.com/docs/2.0/configuration-reference/#schedule
A schedule statement in yaml seems more convenient.

**Comment by bradrydzewski** [comment]
> Lets talk what it would take for real scheduling support, like in CircleCI or Jenkins or Travis or [insert any CI solution but Drone here].

cron support will be available in the 0.9 enterprise edition release. You can learn more about the enterprise edition here https://drone.io/enterprise/

**Comment by tboerger** [comment]
And there are still projects like drone-cron or scheduled jobs within Kubernetes.
